packages/design-system/src/components/query-builder/group.tsx

```diff
@@ -229,27 +241,29 @@ export function RuleGroupHeaderComponents(
           />
         </div>
       )}
-      {!ruleGroup.schema.showCombinatorsBetweenRules &&
-        !ruleGroup.schema.independentCombinators && (
-          <div className={classNames?.group?.combinator}>
-            <CombinatorSelectorControlElement
-              key={TestID.combinators}
-              testID={TestID.combinators}
-              options={ruleGroup.schema.combinators}
-              value={ruleGroup.combinator}
-              title={ruleGroup.translations.combinators.title}
-              className={ruleGroup.classNames.combinators}
-              handleOnChange={ruleGroup.onCombinatorChange}
-              rules={ruleGroup.ruleGroup.rules}
-              level={ruleGroup.path.length}
-              path={ruleGroup.path}
-              disabled={ruleGroup.disabled}
-              context={ruleGroup.context}
-              validation={ruleGroup.validationResult}
-              schema={ruleGroup.schema}
-            />
-          </div>
-        )}
+      {!(
+        ruleGroup.schema.showCombinatorsBetweenRules ||
+        ruleGroup.schema.independentCombinators
+      ) && (
```

Comment:
i don't like this change. i feel like the `!` is sneaky and can be glossed over easily. if other folks agree, i'm happy to revert the change and ignore the lint error instead

Reply:
That isn't the highest priority in that component (for me) for refactor. I'm fine with it this way; I don't have a strong feeling either way in that file as it is.